source/extensions/filters/http/custom_response/policy.h

```diff
@@ -32,6 +31,16 @@ class Policy : public std::enable_shared_from_this<Policy>, public StreamInfo::F
 
 using PolicySharedPtr = std::shared_ptr<const Policy>;
 
+struct CustomResponseFilterState : public std::enable_shared_from_this<CustomResponseFilterState>,
+                                   public StreamInfo::FilterState::Object {
+
+  CustomResponseFilterState(PolicySharedPtr a_policy, absl::optional<::Envoy::Http::Code> code)
+      : policy(a_policy), original_response_code(code) {}
+
+  PolicySharedPtr policy;
+  absl::optional<::Envoy::Http::Code> original_response_code;
```

Comment:
I think that we should make this a general struct `OriginalResponseInfo` so that we can add additional stuff to it (o.g. route, o.g. path, o.g. cluster, etc).

Reply:
I made a change to provide the `ModifyRequestHeadersAction` with response headers, which can potentially add custom filter state.